environments/fetcher/fetcher.go

```diff
@@ -254,6 +254,15 @@ func (fetcher *Fetcher) Fetch(req FetchRequest) (int, error) {
 		if req.FetchType == FETCH_SOURCE {
 			archive = &pkg.Spec.Source
 		} else if req.FetchType == FETCH_DEPLOYMENT {
+			// sometimes, the user may issue curl request to the function even before the source code is built into a deploy pkg.
+			// this results in executor sending a fetch request of type FETCH_DEPLOYMENT and since pkg.Spec.Deployment.Url will be empty,
+			// we hit this "Get : unsupported protocol scheme "" error.
+			// it may be useful to the user if we can send a more meaningful error in such a scenario.
+			if pkg.Status.BuildStatus != fission.BuildStatusSucceeded {
+				e := fmt.Sprintf("Build status for the function's pkg is : %s, can't fetch deployment", pkg.Status.BuildStatus)
+				log.Printf(e)
+				return 500, errors.New(e)
```

Comment:
I think we should give some specific code here, such as 502 or 503.  BTW, does this code make it all the way back through the executor, router, and back to the original request?

Reply:
yeah makes it to router. but you know how router just sends 502 irrespective of the error.
for sample print, you can peek into https://api.travis-ci.org/v3/job/373309088/log.txt, look for below : 
2018/05/01 02:15:00 http: proxy error: Internal error - Build status for the function's pkg is : failed, can't fetch deployment

not sure if 502 or 503 makes sense in this case. i thought about it a lot and decided on 500.